### PR TITLE
Remove fake "browser" and "dom" modules

### PR DESCRIPTION
**Pull request by lucaswoj**
This PR removes the fake "browser" and "dom" modules used in the test environment in favor of using the real modules with jsdom.

This is a follow-up to #3063 

## Checklist

 - [x] briefly describe the changes in this PR
 - [ ] 🍏 
 - [ ] get a PR review :+1: / :-1:

### CODE REVIEW

File: js/util/browser.js [hunk 1]
```diff
@@ -1,29 +1,46 @@
 'use strict';
 
+/**
+ * Unlike js/util/browser.js, this code is written with the expectation
```
Comment (jfirebaugh) [inline js/util/browser.js]: This description was specific to the browser version of browser.js, and can be removed or rewritten.

**Comment by jfirebaugh** [comment]
Can we merge this as-is and go for canvas, ajax, and web_worker in a followup PR? I'd like to write some `Popup` tests that require these changes.

**Comment by lucaswoj** [comment]
@jfirebaugh Sure! I'll fix this up and merge shortly.

**Comment by jfirebaugh** [comment]
The coverage dropped because the test-suite didn't run. This error did not fail the build:

```
/Users/distiller/mapbox-gl-js/test/suite_implementation.js:16
    browser.devicePixelRatio = options.pixelRatio;
                             ^

TypeError: Cannot set property devicePixelRatio of #<Object> which has only a getter
    at module.exports (/Users/distiller/mapbox-gl-js/test/suite_implementation.js:16:30)
    at /Users/distiller/mapbox-gl-js/node_modules/mapbox-gl-test-suite/lib/render.js:62:9
    at /Users/distiller/mapbox-gl-js/node_modules/mapbox-gl-test-suite/lib/harness.js:78:17
    at start (/Users/distiller/mapbox-gl-js/node_modules/mapbox-gl-test-suite/node_modules/queue-async/build/queue.js:35:43)
    at Server.<anonymous> (/Users/distiller/mapbox-gl-js/node_modules/mapbox-gl-test-suite/node_modules/queue-async/build/queue.js:48:24)
    at Server.g (events.js:260:16)
    at emitNone (events.js:67:13)
    at Server.emit (events.js:166:7)
    at emitListeningNT (net.js:1276:10)
    at nextTickCallbackWith1Arg (node.js:431:9)
```

**Comment by lucaswoj** [comment]
Test suite is failing because no fill outlines are being drawn but CI is still passing. This is a fun one :joy:

**Comment by jfirebaugh** [comment]
`set -e` doesn't exit if the command that fails is part of a `&&` list. 😢

**Comment by lucaswoj** [comment]
Easy fix! We weren't using the `&&`.

**Comment by lucaswoj** [comment]
Force pushed to resolve merge conflicts. This should be 🍏  and ready to 🚢 .

**Pull request merged**